woocommerce/woocommerce-android
# Issue/10836 fetch campaigns from new endpoint

Closes: #10836 


### Description

Migrates existing endpoint for fetching Blaze campaigns to new endpoint.

### Testing instructions

1. Log into a site that has at least one Blaze campaign created. 
2. Open Blaze campaign list and check results are displayed correctly. 
3. To test pagination works correctly, apply this patch (NOTE: you need a site that has more than 6 campaigns created in order to trigger pagination upon scrolling): 
```diff
Subject: [PATCH] Reduce page size to trigger pagination on scroll
---
Index: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/campaigs/BlazeCampaignListViewModel.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/campaigs/BlazeCampaignListViewModel.kt b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/campaigs/BlazeCampaignListViewModel.kt
--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/campaigs/BlazeCampaignListViewModel.kt	(revision 05cc0f439be31a38f4da29011cd8c5774f4c4ff5)
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/campaigs/BlazeCampaignListViewModel.kt	(date 1709287809670)
@@ -96,7 +96,7 @@
     }
 
     private suspend fun loadCampaigns(itemsAlreadyLoaded: Int) {
-        val result = blazeCampaignsStore.fetchBlazeCampaigns(selectedSite.get(), itemsAlreadyLoaded)
+        val result = blazeCampaignsStore.fetchBlazeCampaigns(selectedSite.get(), itemsAlreadyLoaded, 6)
         if (result.isError || result.model == null) {
             triggerEvent(Event.ShowSnackbar(R.string.blaze_campaign_list_error_fetching_campaigns))
         } else {

```

Note: To better understand how new pagination system works please refer to https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2963

### Images/gif



https://github.com/woocommerce/woocommerce-android/assets/2663464/01e40e7d-19cf-4a24-a7a4-8142e52c411a
